test/integration/src/main/java/io/pravega/test/integration/compatibility/CompatibilityChecker.java

```diff
@@ -221,7 +232,99 @@ private void checkStreamDelete() {
+                EventWriterConfig.builder().build());
+        // Begin a transaction
+        Transaction<String> txn = writerTxn.beginTxn();
+        assertNotNull(txn.getTxnId());
+        // Writing to the transaction
+        txn.writeEvent("event test");
+        // Checking and validating the Transaction status.
+        Transaction.Status status = txn.checkStatus();
+        assertEquals(Transaction.Status.OPEN, status);
+        // Aborting the transaction
+        txn.abort();
+        // It must fail if we are going to write to an aborted or aborting transaction.
+        assertThrows(TxnFailedException.class, () -> txn.writeEvent("event test"));
+        String readerGroupId = UUID.randomUUID().toString().replace("-", "");
+        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder().stream(Stream.of(scopeName, streamName)).build();
+
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scopeName, controllerURI);
+        readerGroupManager.createReaderGroup(readerGroupId, readerGroupConfig);
+        @Cleanup
+        EventStreamReader<String> reader =  clientFactory.createReader("reader-1", readerGroupId, new UTF8StringSerializer(), ReaderConfig.builder().build());
+        // It should not read an event from the stream because transaction was aborted.
+        EventRead<String>  eventRead =  reader.readNextEvent(READER_TIMEOUT_MS);
+        assertTrue(eventRead.getEvent() == null);
+        assertEquals(Transaction.Status.ABORTED, txn.checkStatus());
+    }
+
+    /**
+     * This method tests the ability to successfully commit a transaction, including writing and reading events from a stream.
+     * @throws TxnFailedException if unable to commit or write to the transaction.
+     */
+    @Test(timeout = 20000)
+    private void checkTransactionReadAndWrite() throws TxnFailedException {
+        String scopeName = "transaction-test-scope";
+        String streamName = "transaction-test-stream";
+        streamManager.createScope(scopeName);
+        streamManager.createStream(scopeName, streamName, streamConfig);
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scopeName, ClientConfig.builder().controllerURI(controllerURI).build());
+        @Cleanup
+        TransactionalEventStreamWriter<String> writerTxn = clientFactory.createTransactionalEventWriter(streamName, new UTF8StringSerializer(),
+                EventWriterConfig.builder().build());
+        assertNotNull(writerTxn);
+        // Begin a transaction.
+        Transaction<String> txn = writerTxn.beginTxn();
+        assertNotNull(txn.getTxnId());
+        // Writing 10 Events to the transaction
+        for (int event = 1; event <= 10; event++) {
+            txn.writeEvent("event test" + event);
+        }
+        // Checking Status of transaction.
+        Transaction.Status status = txn.checkStatus();
+        assertEquals(Transaction.Status.OPEN, status);
```

Comment:
nit: this can be handled in a single line
`assertEquals(Transaction.Status.OPEN, txn.checkStatus());`